### PR TITLE
Switch docker image to use Azul one as base

### PR DESCRIPTION
**Pull request by electrum**
## Description

This reverts #13376.

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: core/docker/build.sh [hunk 1]
```diff
@@ -70,14 +75,15 @@ cp -R default "${WORK_DIR}/"
 TAG_PREFIX="trino:${TRINO_VERSION}"
 
 for arch in "${ARCHITECTURES[@]}"; do
-    echo "🫙  Building the image for $arch"
+    echo "🫙  Building the image for $arch using azul/zulu-openjdk:${AZUL_DOCKER_TAG} as a base"
     docker build \
         "${WORK_DIR}" \
         --pull \
         --platform "linux/$arch" \
```
Comment (nineinchnick) [inline core/docker/build.sh:82]: Looks like they still publish images as `linux/arm64/v8`, not `linux/arm64` as before they broke it.

Comment (electrum) [reply core/docker/build.sh:82]: Is that a problem? It's the same as the `ubuntu` image, for example.

Comment (electrum) [reply core/docker/build.sh:82]: I think this is a different issue

> The requested image's platform (linux/arm64/v8) does not match the detected host platform (linux/amd64) and no specific platform was requested

We'll have to wait for them to fix this before merging.

**Comment by electrum** [comment]
Superseded by #13522

**Pull request closed without merging**